src/SFA.DAS.ApprenticeAccounts/Application/Commands/UpdateAllApprenticePreferencesCommand/UpdateAllApprenticePreferencesCommandHandler.cs

```diff
@@ -0,0 +1,68 @@
+﻿using MediatR;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.ApprenticeAccounts.Data;
+using SFA.DAS.ApprenticeAccounts.Data.Models;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticeAccounts.Application.Commands.UpdateAllApprenticePreferencesCommand
+{
+    public class UpdateAllApprenticePreferencesCommandHandler : IRequestHandler<UpdateAllApprenticePreferencesCommand>
+    {
+        private readonly IApprenticePreferencesContext _apprenticePreferencesContext;
+        private readonly IApprenticeContext _apprenticeContext;
+        private readonly IPreferencesContext _preferencesContext;
+        private readonly ILogger<UpdateAllApprenticePreferencesCommandHandler> _logger;
+
+        public UpdateAllApprenticePreferencesCommandHandler(IApprenticePreferencesContext apprenticePreferencesContext,
+            IApprenticeContext apprenticeContext, IPreferencesContext preferencesContext, ILogger<UpdateAllApprenticePreferencesCommandHandler> logger)
+        {
+            _apprenticePreferencesContext = apprenticePreferencesContext;
+            _apprenticeContext = apprenticeContext;
+            _preferencesContext = preferencesContext;
+            _logger = logger;
+        }
+
+        public async Task<Unit> Handle(UpdateAllApprenticePreferencesCommand request, CancellationToken cancellationToken)
+        {
+            foreach (var apprenticePreference in request.ApprenticePreferences)
+            {
```

Comment:
You could maybe do some more validation here. For example, can this endpoint only be called with one apprentice at a time? or can it be multiple apprentices? If it's the former you can assert that all the preferences in the list have the same ID and only retrieve the apprentice once for example. Something to consider.